modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v2_0/Swagger2Translator.java

```diff
@@ -581,4 +617,523 @@ public static Swagger getSwagger(Definition definition) {
         // Reference to a representation
         return new RefProperty().asDefault(type);
     }
+
+    /**
+     * Translates a Swagger definition to a Restlet Web API Definition
+     * 
+     * @param Swagger
+     *            The translated Swagger 2.0 definition
+     * 
+     * @return The Restlet Web API definition
+     */
+    public static Definition translate(Swagger swagger) {
+
+        // conversion
+        Definition definition = new Definition();
+
+        // fill RWADef main attributes
+        fillRwadefEndpoints(swagger, definition);
+
+        List<String> produces = new ArrayList<>();
+        List<String> consumes = new ArrayList<>();
+        fillRwadefGeneralInformation(swagger, definition, produces, consumes);
+
+        // fill definition.sections
+        Contract contract = definition.getContract();
+        fillSections(swagger, contract);
+
+        // Get declared parameters
+        Map<String, org.restlet.ext.apispark.internal.model.Parameter> parameters = new HashMap<>();
+        fillDeclaredParameters(swagger, definition, parameters);
+
+        // fill definition.representations
+        fillRepresentations(swagger, contract);
+
+        // fill definition.resources
+        fillResources(swagger, contract, produces, consumes, parameters);
+
+        for (Representation representation : contract.getRepresentations()) {
+            representation.addSectionsToProperties(contract);
+        }
+
+        return definition;
+    }
+
+    private static void fillDeclaredParameters(Swagger swagger, Definition definition,
+            Map<String, org.restlet.ext.apispark.internal.model.Parameter> parameters) {
+        if (swagger.getParameters() == null) {
+            return;
+        }
+
+        for (String key : swagger.getParameters().keySet()) {
+            Parameter swaggerParameter = swagger.getParameters().get(key);
+            if (swaggerParameter instanceof QueryParameter) {
+                org.restlet.ext.apispark.internal.model.QueryParameter queryParameter =
+                        new org.restlet.ext.apispark.internal.model.QueryParameter();
+                fillRwadefQueryParameter(queryParameter, (QueryParameter) swaggerParameter);
+                parameters.put(key, queryParameter);
+
+            } else if (swaggerParameter instanceof PathParameter) {
+                org.restlet.ext.apispark.internal.model.PathVariable pathVariable =
+                        new org.restlet.ext.apispark.internal.model.PathVariable();
+                fillRwadefPathVariable(pathVariable, (PathParameter) swaggerParameter);
+                parameters.put(key, pathVariable);
+
+            } else if (swaggerParameter instanceof HeaderParameter) {
+                Header header = new Header();
+                fillRwadefHeader(header, (HeaderParameter) swaggerParameter);
+                parameters.put(key, header);
+
+            }
```

Comment:
Done 
There should only be four possible types, Body, Query, Header or PathParameter.